src/libopensc/pkcs15-prkey.c

```diff
@@ -258,10 +258,6 @@ int sc_pkcs15_decode_prkdf_entry(struct sc_pkcs15_card *p15card,
 	memset(gostr3410_params, 0, sizeof(gostr3410_params));
 
 	r = sc_asn1_decode_choice(ctx, asn1_prkey, *buf, *buflen, buf, buflen);
-	if (r < 0) {
-		/* This might have allocated something. If so, clear it now */
-		free(info.subject.value);
```

Comment:
Hmm, where is this freed? Do you mean `sc_pkcs15_parse_df()`?

Reply:
My bad. Thank you for double-checking. I though that this function is more similar to the one we talked about previously (`sc_pkcs15_decode_pukdf_entry()`, but now I see this one uses static allocation and then memcpy so this is actually needed.
To avoid later confusion, I would suggest to rewrite either of these functions so they are more similar in the internal semantics, probably the prkey one as it allocates as late as possible, making the whole function more readable and avoiding `goto`. What do you think? I will push a new commit.

Reply:
After checking a bit more, the functions are still different in spite of allocations of other data so unless you would be strongly for refactoring one of them, I would leave them as they are now.